index.js

```diff
@@ -36,10 +36,17 @@ function isMatch(pattern, options) {
     };
   }
 
-  if (Array.isArray(pattern) || isObject(pattern)) {
+  if (Array.isArray(pattern)) {
+    return function (val) {
+      return mm(val, pattern, options).length !== 0;
```

Comment:
should we use `mm.any(val, pattern, options)` here? I believe `mm.any()` exits early when a match is found, so it would be faster here...